src/Core/Services/StateService.cs

```diff
@@ -1001,17 +1001,25 @@ public async Task SetPushInitialPromptShownAsync(bool? value)
             await SetValueAsync(key, value, options);
         }
 
-        public async Task<DateTime?> GetPushLastRegistrationDateAsync()
+        public async Task<DateTime?> GetPushLastRegistrationDateAsync(string userId = null)
         {
+            if (userId == null)
+            {
+                userId = await GetActiveUserIdAsync();
+            }
             var options = await GetDefaultStorageOptionsAsync();
-            var key = Constants.PushLastRegistrationDateKey;
+            var key = Constants.PushLastRegistrationDateKey(userId);
             return await GetValueAsync<DateTime?>(key, options);
         }
 
-        public async Task SetPushLastRegistrationDateAsync(DateTime? value)
+        public async Task SetPushLastRegistrationDateAsync(DateTime? value, string userId = null)
```

Comment:
And here's the pattern for account-specific setters:

`var reconciledOptions = ReconcileOptions(new StorageOptions { UserId = userId }, await GetDefaultStorageOptionsAsync());`
`var key = Constants.PushLastRegistrationDateKey(reconciledOptions.UserId);`
`await SetValueAsync(key, value, reconciledOptions);`